flickit-assessment-kit/src/main/java/org/flickit/assessment/kit/adapter/out/persistence/kitversion/KitVersionMapper.java

```diff
@@ -35,4 +36,8 @@ public static KitVersion mapToDomainModel(KitVersionJpaEntity entity) {
             entity.getCreatedBy(),
             entity.getLastModifiedBy());
     }
+
+    public static long generateStatusVersion(KitVersionStatus status, long kitVersionId) {
+        return status == KitVersionStatus.ARCHIVE ? -kitVersionId : status.getId();
```

Comment:
**Suggestion:** Refactor the `generateStatusVersion` method to ensure it handles all possible `KitVersionStatus` values correctly, avoiding potential misuse of the method which currently only explicitly checks for `ARCHIVE`. [general, importance: 6]
```suggestion
        public static long generateStatusVersion(KitVersionStatus status, long kitVersionId) {
            switch (status) {
                case ARCHIVE:
                    return -kitVersionId;
                case ACTIVE:
                    return kitVersionId + 1000;  // Example adjustment
                default:
                    return status.getId();
            }
        }
```